src/internal/log/log.go

```diff
@@ -55,6 +55,22 @@ func Pretty(entry *logrus.Entry) ([]byte, error) {
 	return serialized, nil
 }
 
+var jsonFormatter = &logrus.JSONFormatter{}
+
+// JSONPretty is similar to Pretty() above, but it formats logrus log entries as
+// valid JSON objects. This is required by GCP (or else it misunderstands
+// severity and treats all log lines as ERROR logs).
+func JSONPretty(entry *logrus.Entry) ([]byte, error) {
```

Comment:
Not sure what field names JSONFormatter uses out of the box, but Google wants "severity", "message", "time" (in RFC3339).  To solve this particular customer's issue, the log level being "severity" is probably the most important.

If we want to go crazy, here's everything they support: https://cloud.google.com/logging/docs/agent/logging/configuration

Reply:
```go
	log.SetFormatter(&log.JSONFormatter{
		FieldMap: log.FieldMap{
			log.FieldKeyTime:  "time",
			log.FieldKeyLevel: "severity",
			log.FieldKeyMsg:   "message",
		},
		// https://github.com/sirupsen/logrus/pull/162/files
		TimestampFormat: time.RFC3339Nano,
	})
```
is how hub and helium do it

Reply:
Done—thanks for the tip!